packages/web/src/pages/upload-page/forms/AttributionModalForm.tsx

```diff
@@ -0,0 +1,317 @@
+        description={messages.aiGenerated.description}
+      >
+        <AiAttributionDropdown
+          {...aiUserIdField}
+          onSelect={(value: AttributionFormValues[typeof AI_USER_ID]) =>
+            setAiUserId(value)
+          }
+        />
+      </SwitchRowField>
+      <Divider />
+      <div className={cn(styles.col, styles.gap4)}>
+        <div
+          className={typeStyles.titleLarge}
+        >{`${messages.isrc.header} / ${messages.iswc.header}`}</div>
+        <span className={cn(styles.row, styles.gap6)}>
+          <InputV2
+            {...isrcField}
+            variant={InputV2Variant.ELEVATED_PLACEHOLDER}
+            label={messages.isrc.header}
+            placeholder={messages.isrc.placeholder}
+          />
+          <InputV2
+            {...iswcField}
+            variant={InputV2Variant.ELEVATED_PLACEHOLDER}
+            label={messages.iswc.header}
+            placeholder={messages.iswc.placeholder}
+          />
+        </span>
+      </div>
+      <Divider />
+      <div className={cn(styles.col, styles.gap6)}>
+        <div className={typeStyles.titleLarge}>{messages.licenseType}</div>
+        <div className={styles.attributionCommercialRow}>
+          <div
+            className={cn(styles.attributionRowItem, styles.col, styles.gap2)}
+          >
+            <div className={typeStyles.titleMedium}>
+              {messages.allowAttribution.header}
+            </div>
+            <SegmentedControl
+              defaultSelected={allowAttribution}
+              // @ts-ignore boolean support works
+              options={allowAttributionValues}
+              // @ts-ignore
+              onSelectOption={setAllowAttribution}
+            />
+          </div>
+          <Divider className={styles.verticalDivider} type='vertical' />
+          <div
+            className={cn(styles.attributionRowItem, styles.col, styles.gap2, {
+              [styles.disabled]: !allowAttribution
+            })}
+          >
+            <div className={typeStyles.titleMedium}>
+              {messages.commercialUse.header}
+            </div>
+            <SegmentedControl
+              fullWidth
+              defaultSelected={commercialUse}
+              // @ts-ignore boolean support works
+              options={commercialUseValues}
+              // @ts-ignore
+              onSelectOption={setCommercialUse}
+              disabled={!allowAttribution}
+            />
+          </div>
+        </div>
+        <div className={cn(styles.col, styles.gap2)}>
+          <div
+            className={cn(typeStyles.titleMedium, {
+              [styles.disabled]: !allowAttribution
+            })}
+          >
+            {messages.derivativeWorks.header}
+          </div>
+          <SegmentedControl
+            fullWidth
+            defaultSelected={derivativeWorks}
+            // @ts-ignore boolean support works
+            options={derivativeWorksValues}
+            // @ts-ignore
```

Comment:
Pretty annoying that we have to do this

Reply:
Yeah it's fixable but involves rewriting a bunch of stuff I think. I tried fooling with the types, but we'd have to ts-ignore in the component instead. Maybe better? idk